force-app/main/default/lwc/picklist/picklist.js

```diff
@@ -22,9 +22,12 @@ export default class Picklist extends LightningElement {
         }
 
         this._options = this.picklist.picklistValues.map(picklistValue => {
-            // TODO: Set defaultValue if value not provided?
             let isSelected = this.value && this.value.includes(picklistValue.value);
 
+            if (isSelected && this.value === picklistValue.value) {
+                this.value = picklistValue.value;
```

Comment:
I think this might still not be what you were going after, let's discuss Monday.